process/transaction/interceptedTransaction.go

```diff
@@ -121,30 +121,26 @@ func (inTx *InterceptedTransaction) processFields(txBuffWithSig []byte) ([]byte,
 		inTx.rcvShard = inTx.sndShard
 	}
 
-	inTx.isAddressedToOtherShards = inTx.rcvShard != inTx.coordinator.SelfId() &&
-		inTx.sndShard != inTx.coordinator.SelfId()
+	inTx.isForMyShard = inTx.rcvShard == inTx.coordinator.SelfId() ||
```

Comment:
You could separate in 2 variables (isMyShardRecv and isMyShardSender), and then apply || on these two variables.

Reply:
Agree. Done